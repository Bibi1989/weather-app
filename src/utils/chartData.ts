import _ from "lodash";
import { FormatedWeatherInterface } from "typescript/weather.types";

export const returnChartDataAndLabel = (
  charts: FormatedWeatherInterface[]
) => ({
  data: _.map(charts, (chart) => chart.temperature),
  label: _.map(charts, (chart) => chart.time),
});
