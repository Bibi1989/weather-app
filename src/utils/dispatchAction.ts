import { Dispatch } from "redux";

import {
  getWeathersFromYourLocation,
  getWeathers,
} from "redux/slices/weathers/weatherActions";
import { CoordInterface } from "typescript/weather.types";

export const dispatchActions = (
  dispatch: Dispatch<any>,
  units: string,
  coord: CoordInterface,
  condition: boolean
) => {
  if (condition) {
    dispatch(
      getWeathersFromYourLocation({
        units,
        coord: { long: coord.long, lat: coord.lat },
      })
    );
  } else {
    dispatch(getWeathers({ units }));
  }
};
