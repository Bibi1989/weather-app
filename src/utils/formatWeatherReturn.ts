import { WeatherInterface } from "typescript/weather.types";

export const formatWeatherReturn = (
  weathers: WeatherInterface[],
  units: string
) => {
  return weathers?.map((weather: WeatherInterface) => ({
    temperature: weather?.main?.temp,
    dt_txt: weather?.dt_txt,
    time: weather?.dt_txt.split(" ")[1],
    dt: weather?.dt,
    icon: weather?.weather[0]?.icon,
    description: weather?.weather[0]?.description,
    main: weather?.weather[0]?.main,
    units,
  }));
};
