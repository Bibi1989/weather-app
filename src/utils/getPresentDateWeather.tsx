import _ from "lodash";
import moment from "moment";

import { FormatedWeatherInterface } from "typescript/weather.types";

export const getDateWeatherForcast = (
  weathers: FormatedWeatherInterface[],
  date: string
) => {
  const now = date;
  return _.filter(
    weathers,
    (weather: FormatedWeatherInterface) =>
      getJustDate(weather?.dt_txt) === getJustDate(now)
  );
};

export const getPresentDayWeather = (weathers: FormatedWeatherInterface[]) => {
  return _.uniqBy(
    _.map(weathers, (weather: FormatedWeatherInterface) => ({
      ...weather,
      date: getJustDate(weather?.dt_txt),
    })),
    "date"
  );
};

export const formatDate = (date: string): string => {
  const dateString = moment(date).format("D MMM YYYY");
  return dateString;
};

export const getJustDate = (date: string): string => {
  return date.split(" ")[0];
};

export const getDay = (date: string): string => {
  const dateString = new Date(date).toDateString();
  const day = dateString.split(" ")[2];
  return day;
};
