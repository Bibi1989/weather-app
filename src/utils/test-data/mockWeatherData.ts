export const mockWeatherData = {
  cod: "200",
  message: 0,
  cnt: 40,
  list: [
    {
      dt: 1637377200,
      main: {
        temp: 275.82,
        feels_like: 275.82,
        temp_min: 275.44,
        temp_max: 275.82,
        pressure: 1031,
        sea_level: 1031,
        grnd_level: 966,
        humidity: 92,
        temp_kf: 0.38,
      },
      weather: [
        {
          id: 801,
          main: "Clouds",
          description: "few clouds",
          icon: "02n",
        },
      ],
      clouds: {
        all: 17,
      },
      wind: {
        speed: 0.57,
        deg: 183,
        gust: 0.65,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-20 03:00:00",
    },
    {
      dt: 1637388000,
      main: {
        temp: 275.4,
        feels_like: 275.4,
        temp_min: 275.1,
        temp_max: 275.4,
        pressure: 1030,
        sea_level: 1030,
        grnd_level: 965,
        humidity: 93,
        temp_kf: 0.3,
      },
      weather: [
        {
          id: 801,
          main: "Clouds",
          description: "few clouds",
          icon: "02n",
        },
      ],
      clouds: {
        all: 17,
      },
      wind: {
        speed: 1.01,
        deg: 172,
        gust: 1.03,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-20 06:00:00",
    },
    {
      dt: 1637398800,
      main: {
        temp: 278.88,
        feels_like: 278.88,
        temp_min: 278.88,
        temp_max: 278.88,
        pressure: 1028,
        sea_level: 1028,
        grnd_level: 965,
        humidity: 78,
        temp_kf: 0,
      },
      weather: [
        {
          id: 801,
          main: "Clouds",
          description: "few clouds",
          icon: "02d",
        },
      ],
      clouds: {
        all: 18,
      },
      wind: {
        speed: 0.98,
        deg: 124,
        gust: 1.33,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "d",
      },
      dt_txt: "2021-11-20 09:00:00",
    },
    {
      dt: 1637409600,
      main: {
        temp: 282.35,
        feels_like: 282.35,
        temp_min: 282.35,
        temp_max: 282.35,
        pressure: 1025,
        sea_level: 1025,
        grnd_level: 962,
        humidity: 66,
        temp_kf: 0,
      },
      weather: [
        {
          id: 801,
          main: "Clouds",
          description: "few clouds",
          icon: "02d",
        },
      ],
      clouds: {
        all: 14,
      },
      wind: {
        speed: 0.88,
        deg: 60,
        gust: 0.76,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "d",
      },
      dt_txt: "2021-11-20 12:00:00",
    },
    {
      dt: 1637420400,
      main: {
        temp: 279.62,
        feels_like: 279.62,
        temp_min: 279.62,
        temp_max: 279.62,
        pressure: 1022,
        sea_level: 1022,
        grnd_level: 959,
        humidity: 82,
        temp_kf: 0,
      },
      weather: [
        {
          id: 801,
          main: "Clouds",
          description: "few clouds",
          icon: "02d",
        },
      ],
      clouds: {
        all: 11,
      },
      wind: {
        speed: 0.77,
        deg: 87,
        gust: 0.8,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "d",
      },
      dt_txt: "2021-11-20 15:00:00",
    },
    {
      dt: 1637431200,
      main: {
        temp: 277.26,
        feels_like: 277.26,
        temp_min: 277.26,
        temp_max: 277.26,
        pressure: 1022,
        sea_level: 1022,
        grnd_level: 958,
        humidity: 88,
        temp_kf: 0,
      },
      weather: [
        {
          id: 801,
          main: "Clouds",
          description: "few clouds",
          icon: "02n",
        },
      ],
      clouds: {
        all: 11,
      },
      wind: {
        speed: 0.93,
        deg: 203,
        gust: 0.93,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-20 18:00:00",
    },
    {
      dt: 1637442000,
      main: {
        temp: 276.56,
        feels_like: 274.96,
        temp_min: 276.56,
        temp_max: 276.56,
        pressure: 1020,
        sea_level: 1020,
        grnd_level: 957,
        humidity: 89,
        temp_kf: 0,
      },
      weather: [
        {
          id: 802,
          main: "Clouds",
          description: "scattered clouds",
          icon: "03n",
        },
      ],
      clouds: {
        all: 28,
      },
      wind: {
        speed: 1.75,
        deg: 194,
        gust: 1.71,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-20 21:00:00",
    },
    {
      dt: 1637452800,
      main: {
        temp: 276.1,
        feels_like: 274.49,
        temp_min: 276.1,
        temp_max: 276.1,
        pressure: 1018,
        sea_level: 1018,
        grnd_level: 954,
        humidity: 89,
        temp_kf: 0,
      },
      weather: [
        {
          id: 801,
          main: "Clouds",
          description: "few clouds",
          icon: "02n",
        },
      ],
      clouds: {
        all: 19,
      },
      wind: {
        speed: 1.7,
        deg: 206,
        gust: 1.66,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-21 00:00:00",
    },
    {
      dt: 1637463600,
      main: {
        temp: 275.86,
        feels_like: 274.09,
        temp_min: 275.86,
        temp_max: 275.86,
        pressure: 1016,
        sea_level: 1016,
        grnd_level: 952,
        humidity: 88,
        temp_kf: 0,
      },
      weather: [
        {
          id: 800,
          main: "Clear",
          description: "clear sky",
          icon: "01n",
        },
      ],
      clouds: {
        all: 10,
      },
      wind: {
        speed: 1.8,
        deg: 230,
        gust: 1.79,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-21 03:00:00",
    },
    {
      dt: 1637474400,
      main: {
        temp: 275.47,
        feels_like: 273.09,
        temp_min: 275.47,
        temp_max: 275.47,
        pressure: 1015,
        sea_level: 1015,
        grnd_level: 951,
        humidity: 88,
        temp_kf: 0,
      },
      weather: [
        {
          id: 801,
          main: "Clouds",
          description: "few clouds",
          icon: "02n",
        },
      ],
      clouds: {
        all: 11,
      },
      wind: {
        speed: 2.28,
        deg: 229,
        gust: 3.43,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-21 06:00:00",
    },
    {
      dt: 1637485200,
      main: {
        temp: 279.46,
        feels_like: 277.26,
        temp_min: 279.46,
        temp_max: 279.46,
        pressure: 1013,
        sea_level: 1013,
        grnd_level: 951,
        humidity: 71,
        temp_kf: 0,
      },
      weather: [
        {
          id: 800,
          main: "Clear",
          description: "clear sky",
          icon: "01d",
        },
      ],
      clouds: {
        all: 6,
      },
      wind: {
        speed: 2.96,
        deg: 264,
        gust: 9.02,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "d",
      },
      dt_txt: "2021-11-21 09:00:00",
    },
    {
      dt: 1637496000,
      main: {
        temp: 282.56,
        feels_like: 280.7,
        temp_min: 282.56,
        temp_max: 282.56,
        pressure: 1011,
        sea_level: 1011,
        grnd_level: 949,
        humidity: 60,
        temp_kf: 0,
      },
      weather: [
        {
          id: 800,
          main: "Clear",
          description: "clear sky",
          icon: "01d",
        },
      ],
      clouds: {
        all: 5,
      },
      wind: {
        speed: 3.46,
        deg: 284,
        gust: 6.7,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "d",
      },
      dt_txt: "2021-11-21 12:00:00",
    },
    {
      dt: 1637506800,
      main: {
        temp: 279.48,
        feels_like: 278.36,
        temp_min: 279.48,
        temp_max: 279.48,
        pressure: 1011,
        sea_level: 1011,
        grnd_level: 948,
        humidity: 78,
        temp_kf: 0,
      },
      weather: [
        {
          id: 800,
          main: "Clear",
          description: "clear sky",
          icon: "01d",
        },
      ],
      clouds: {
        all: 6,
      },
      wind: {
        speed: 1.72,
        deg: 305,
        gust: 1.91,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "d",
      },
      dt_txt: "2021-11-21 15:00:00",
    },
    {
      dt: 1637517600,
      main: {
        temp: 276.71,
        feels_like: 276.71,
        temp_min: 276.71,
        temp_max: 276.71,
        pressure: 1012,
        sea_level: 1012,
        grnd_level: 949,
        humidity: 89,
        temp_kf: 0,
      },
      weather: [
        {
          id: 800,
          main: "Clear",
          description: "clear sky",
          icon: "01n",
        },
      ],
      clouds: {
        all: 4,
      },
      wind: {
        speed: 0.88,
        deg: 299,
        gust: 1.05,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-21 18:00:00",
    },
    {
      dt: 1637528400,
      main: {
        temp: 276.3,
        feels_like: 276.3,
        temp_min: 276.3,
        temp_max: 276.3,
        pressure: 1013,
        sea_level: 1013,
        grnd_level: 950,
        humidity: 91,
        temp_kf: 0,
      },
      weather: [
        {
          id: 803,
          main: "Clouds",
          description: "broken clouds",
          icon: "04n",
        },
      ],
      clouds: {
        all: 75,
      },
      wind: {
        speed: 0.74,
        deg: 299,
        gust: 0.8,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-21 21:00:00",
    },
    {
      dt: 1637539200,
      main: {
        temp: 277.19,
        feels_like: 277.19,
        temp_min: 277.19,
        temp_max: 277.19,
        pressure: 1013,
        sea_level: 1013,
        grnd_level: 950,
        humidity: 89,
        temp_kf: 0,
      },
      weather: [
        {
          id: 804,
          main: "Clouds",
          description: "overcast clouds",
          icon: "04n",
        },
      ],
      clouds: {
        all: 88,
      },
      wind: {
        speed: 0.34,
        deg: 332,
        gust: 0.73,
      },
      visibility: 10000,
      pop: 0.02,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-22 00:00:00",
    },
    {
      dt: 1637550000,
      main: {
        temp: 277.13,
        feels_like: 277.13,
        temp_min: 277.13,
        temp_max: 277.13,
        pressure: 1013,
        sea_level: 1013,
        grnd_level: 950,
        humidity: 91,
        temp_kf: 0,
      },
      weather: [
        {
          id: 804,
          main: "Clouds",
          description: "overcast clouds",
          icon: "04n",
        },
      ],
      clouds: {
        all: 100,
      },
      wind: {
        speed: 1.14,
        deg: 29,
        gust: 1.6,
      },
      visibility: 10000,
      pop: 0.05,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-22 03:00:00",
    },
    {
      dt: 1637560800,
      main: {
        temp: 277.18,
        feels_like: 275.55,
        temp_min: 277.18,
        temp_max: 277.18,
        pressure: 1014,
        sea_level: 1014,
        grnd_level: 951,
        humidity: 94,
        temp_kf: 0,
      },
      weather: [
        {
          id: 804,
          main: "Clouds",
          description: "overcast clouds",
          icon: "04n",
        },
      ],
      clouds: {
        all: 100,
      },
      wind: {
        speed: 1.85,
        deg: 22,
        gust: 3.66,
      },
      visibility: 10000,
      pop: 0.05,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-22 06:00:00",
    },
    {
      dt: 1637571600,
      main: {
        temp: 277.19,
        feels_like: 274.64,
        temp_min: 277.19,
        temp_max: 277.19,
        pressure: 1016,
        sea_level: 1016,
        grnd_level: 953,
        humidity: 95,
        temp_kf: 0,
      },
      weather: [
        {
          id: 500,
          main: "Rain",
          description: "light rain",
          icon: "10d",
        },
      ],
      clouds: {
        all: 100,
      },
      wind: {
        speed: 2.83,
        deg: 27,
        gust: 5.62,
      },
      visibility: 8846,
      pop: 0.38,
      rain: {
        "3h": 0.13,
      },
      sys: {
        pod: "d",
      },
      dt_txt: "2021-11-22 09:00:00",
    },
    {
      dt: 1637582400,
      main: {
        temp: 277.27,
        feels_like: 274.51,
        temp_min: 277.27,
        temp_max: 277.27,
        pressure: 1017,
        sea_level: 1017,
        grnd_level: 954,
        humidity: 96,
        temp_kf: 0,
      },
      weather: [
        {
          id: 500,
          main: "Rain",
          description: "light rain",
          icon: "10d",
        },
      ],
      clouds: {
        all: 100,
      },
      wind: {
        speed: 3.12,
        deg: 33,
        gust: 5.8,
      },
      visibility: 751,
      pop: 0.3,
      rain: {
        "3h": 0.19,
      },
      sys: {
        pod: "d",
      },
      dt_txt: "2021-11-22 12:00:00",
    },
    {
      dt: 1637593200,
      main: {
        temp: 277.11,
        feels_like: 274.27,
        temp_min: 277.11,
        temp_max: 277.11,
        pressure: 1019,
        sea_level: 1019,
        grnd_level: 955,
        humidity: 96,
        temp_kf: 0,
      },
      weather: [
        {
          id: 500,
          main: "Rain",
          description: "light rain",
          icon: "10d",
        },
      ],
      clouds: {
        all: 100,
      },
      wind: {
        speed: 3.19,
        deg: 36,
        gust: 6.25,
      },
      visibility: 10000,
      pop: 0.24,
      rain: {
        "3h": 0.22,
      },
      sys: {
        pod: "d",
      },
      dt_txt: "2021-11-22 15:00:00",
    },
    {
      dt: 1637604000,
      main: {
        temp: 276.93,
        feels_like: 273.43,
        temp_min: 276.93,
        temp_max: 276.93,
        pressure: 1021,
        sea_level: 1021,
        grnd_level: 957,
        humidity: 96,
        temp_kf: 0,
      },
      weather: [
        {
          id: 500,
          main: "Rain",
          description: "light rain",
          icon: "10n",
        },
      ],
      clouds: {
        all: 100,
      },
      wind: {
        speed: 4.13,
        deg: 53,
        gust: 7.43,
      },
      visibility: 8412,
      pop: 0.6,
      rain: {
        "3h": 0.52,
      },
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-22 18:00:00",
    },
    {
      dt: 1637614800,
      main: {
        temp: 276.15,
        feels_like: 272.39,
        temp_min: 276.15,
        temp_max: 276.15,
        pressure: 1023,
        sea_level: 1023,
        grnd_level: 959,
        humidity: 91,
        temp_kf: 0,
      },
      weather: [
        {
          id: 500,
          main: "Rain",
          description: "light rain",
          icon: "10n",
        },
      ],
      clouds: {
        all: 100,
      },
      wind: {
        speed: 4.26,
        deg: 53,
        gust: 9.72,
      },
      visibility: 9796,
      pop: 0.52,
      rain: {
        "3h": 0.15,
      },
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-22 21:00:00",
    },
    {
      dt: 1637625600,
      main: {
        temp: 275.02,
        feels_like: 271.25,
        temp_min: 275.02,
        temp_max: 275.02,
        pressure: 1024,
        sea_level: 1024,
        grnd_level: 960,
        humidity: 89,
        temp_kf: 0,
      },
      weather: [
        {
          id: 804,
          main: "Clouds",
          description: "overcast clouds",
          icon: "04n",
        },
      ],
      clouds: {
        all: 100,
      },
      wind: {
        speed: 3.87,
        deg: 56,
        gust: 10.02,
      },
      visibility: 10000,
      pop: 0.28,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-23 00:00:00",
    },
    {
      dt: 1637636400,
      main: {
        temp: 274.95,
        feels_like: 270.91,
        temp_min: 274.95,
        temp_max: 274.95,
        pressure: 1025,
        sea_level: 1025,
        grnd_level: 960,
        humidity: 85,
        temp_kf: 0,
      },
      weather: [
        {
          id: 804,
          main: "Clouds",
          description: "overcast clouds",
          icon: "04n",
        },
      ],
      clouds: {
        all: 100,
      },
      wind: {
        speed: 4.27,
        deg: 65,
        gust: 11.26,
      },
      visibility: 10000,
      pop: 0.09,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-23 03:00:00",
    },
    {
      dt: 1637647200,
      main: {
        temp: 274.75,
        feels_like: 270.16,
        temp_min: 274.75,
        temp_max: 274.75,
        pressure: 1026,
        sea_level: 1026,
        grnd_level: 961,
        humidity: 84,
        temp_kf: 0,
      },
      weather: [
        {
          id: 804,
          main: "Clouds",
          description: "overcast clouds",
          icon: "04n",
        },
      ],
      clouds: {
        all: 100,
      },
      wind: {
        speed: 5.14,
        deg: 73,
        gust: 12.46,
      },
      visibility: 10000,
      pop: 0.09,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-23 06:00:00",
    },
    {
      dt: 1637658000,
      main: {
        temp: 275.32,
        feels_like: 270.63,
        temp_min: 275.32,
        temp_max: 275.32,
        pressure: 1026,
        sea_level: 1026,
        grnd_level: 962,
        humidity: 79,
        temp_kf: 0,
      },
      weather: [
        {
          id: 803,
          main: "Clouds",
          description: "broken clouds",
          icon: "04d",
        },
      ],
      clouds: {
        all: 61,
      },
      wind: {
        speed: 5.63,
        deg: 74,
        gust: 12.08,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "d",
      },
      dt_txt: "2021-11-23 09:00:00",
    },
    {
      dt: 1637668800,
      main: {
        temp: 277,
        feels_like: 272.47,
        temp_min: 277,
        temp_max: 277,
        pressure: 1026,
        sea_level: 1026,
        grnd_level: 962,
        humidity: 69,
        temp_kf: 0,
      },
      weather: [
        {
          id: 803,
          main: "Clouds",
          description: "broken clouds",
          icon: "04d",
        },
      ],
      clouds: {
        all: 60,
      },
      wind: {
        speed: 6.28,
        deg: 78,
        gust: 10.31,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "d",
      },
      dt_txt: "2021-11-23 12:00:00",
    },
    {
      dt: 1637679600,
      main: {
        temp: 275.97,
        feels_like: 271.48,
        temp_min: 275.97,
        temp_max: 275.97,
        pressure: 1026,
        sea_level: 1026,
        grnd_level: 962,
        humidity: 74,
        temp_kf: 0,
      },
      weather: [
        {
          id: 803,
          main: "Clouds",
          description: "broken clouds",
          icon: "04d",
        },
      ],
      clouds: {
        all: 53,
      },
      wind: {
        speed: 5.56,
        deg: 77,
        gust: 10.26,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "d",
      },
      dt_txt: "2021-11-23 15:00:00",
    },
    {
      dt: 1637690400,
      main: {
        temp: 273.96,
        feels_like: 268.91,
        temp_min: 273.96,
        temp_max: 273.96,
        pressure: 1026,
        sea_level: 1026,
        grnd_level: 962,
        humidity: 79,
        temp_kf: 0,
      },
      weather: [
        {
          id: 803,
          main: "Clouds",
          description: "broken clouds",
          icon: "04n",
        },
      ],
      clouds: {
        all: 52,
      },
      wind: {
        speed: 5.64,
        deg: 75,
        gust: 13.59,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-23 18:00:00",
    },
    {
      dt: 1637701200,
      main: {
        temp: 272.99,
        feels_like: 267.81,
        temp_min: 272.99,
        temp_max: 272.99,
        pressure: 1026,
        sea_level: 1026,
        grnd_level: 962,
        humidity: 86,
        temp_kf: 0,
      },
      weather: [
        {
          id: 801,
          main: "Clouds",
          description: "few clouds",
          icon: "02n",
        },
      ],
      clouds: {
        all: 20,
      },
      wind: {
        speed: 5.4,
        deg: 82,
        gust: 12.66,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-23 21:00:00",
    },
    {
      dt: 1637712000,
      main: {
        temp: 272.23,
        feels_like: 267.51,
        temp_min: 272.23,
        temp_max: 272.23,
        pressure: 1026,
        sea_level: 1026,
        grnd_level: 961,
        humidity: 89,
        temp_kf: 0,
      },
      weather: [
        {
          id: 801,
          main: "Clouds",
          description: "few clouds",
          icon: "02n",
        },
      ],
      clouds: {
        all: 22,
      },
      wind: {
        speed: 4.33,
        deg: 77,
        gust: 10.57,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-24 00:00:00",
    },
    {
      dt: 1637722800,
      main: {
        temp: 271.65,
        feels_like: 267.45,
        temp_min: 271.65,
        temp_max: 271.65,
        pressure: 1025,
        sea_level: 1025,
        grnd_level: 960,
        humidity: 92,
        temp_kf: 0,
      },
      weather: [
        {
          id: 802,
          main: "Clouds",
          description: "scattered clouds",
          icon: "03n",
        },
      ],
      clouds: {
        all: 42,
      },
      wind: {
        speed: 3.45,
        deg: 78,
        gust: 8.77,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-24 03:00:00",
    },
    {
      dt: 1637733600,
      main: {
        temp: 271.56,
        feels_like: 268.28,
        temp_min: 271.56,
        temp_max: 271.56,
        pressure: 1024,
        sea_level: 1024,
        grnd_level: 959,
        humidity: 92,
        temp_kf: 0,
      },
      weather: [
        {
          id: 803,
          main: "Clouds",
          description: "broken clouds",
          icon: "04n",
        },
      ],
      clouds: {
        all: 53,
      },
      wind: {
        speed: 2.47,
        deg: 75,
        gust: 5.38,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-24 06:00:00",
    },
    {
      dt: 1637744400,
      main: {
        temp: 273.58,
        feels_like: 270.95,
        temp_min: 273.58,
        temp_max: 273.58,
        pressure: 1023,
        sea_level: 1023,
        grnd_level: 959,
        humidity: 79,
        temp_kf: 0,
      },
      weather: [
        {
          id: 803,
          main: "Clouds",
          description: "broken clouds",
          icon: "04d",
        },
      ],
      clouds: {
        all: 59,
      },
      wind: {
        speed: 2.21,
        deg: 78,
        gust: 3.23,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "d",
      },
      dt_txt: "2021-11-24 09:00:00",
    },
    {
      dt: 1637755200,
      main: {
        temp: 277.29,
        feels_like: 275.65,
        temp_min: 277.29,
        temp_max: 277.29,
        pressure: 1021,
        sea_level: 1021,
        grnd_level: 957,
        humidity: 65,
        temp_kf: 0,
      },
      weather: [
        {
          id: 802,
          main: "Clouds",
          description: "scattered clouds",
          icon: "03d",
        },
      ],
      clouds: {
        all: 37,
      },
      wind: {
        speed: 1.88,
        deg: 71,
        gust: 2.51,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "d",
      },
      dt_txt: "2021-11-24 12:00:00",
    },
    {
      dt: 1637766000,
      main: {
        temp: 275.54,
        feels_like: 273.69,
        temp_min: 275.54,
        temp_max: 275.54,
        pressure: 1020,
        sea_level: 1020,
        grnd_level: 956,
        humidity: 77,
        temp_kf: 0,
      },
      weather: [
        {
          id: 800,
          main: "Clear",
          description: "clear sky",
          icon: "01d",
        },
      ],
      clouds: {
        all: 6,
      },
      wind: {
        speed: 1.82,
        deg: 61,
        gust: 2.09,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "d",
      },
      dt_txt: "2021-11-24 15:00:00",
    },
    {
      dt: 1637776800,
      main: {
        temp: 273.28,
        feels_like: 273.28,
        temp_min: 273.28,
        temp_max: 273.28,
        pressure: 1020,
        sea_level: 1020,
        grnd_level: 955,
        humidity: 87,
        temp_kf: 0,
      },
      weather: [
        {
          id: 800,
          main: "Clear",
          description: "clear sky",
          icon: "01n",
        },
      ],
      clouds: {
        all: 6,
      },
      wind: {
        speed: 1.1,
        deg: 104,
        gust: 1.17,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-24 18:00:00",
    },
    {
      dt: 1637787600,
      main: {
        temp: 272.87,
        feels_like: 272.87,
        temp_min: 272.87,
        temp_max: 272.87,
        pressure: 1019,
        sea_level: 1019,
        grnd_level: 954,
        humidity: 89,
        temp_kf: 0,
      },
      weather: [
        {
          id: 800,
          main: "Clear",
          description: "clear sky",
          icon: "01n",
        },
      ],
      clouds: {
        all: 8,
      },
      wind: {
        speed: 0.93,
        deg: 113,
        gust: 0.97,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-24 21:00:00",
    },
    {
      dt: 1637798400,
      main: {
        temp: 272.68,
        feels_like: 272.68,
        temp_min: 272.68,
        temp_max: 272.68,
        pressure: 1017,
        sea_level: 1017,
        grnd_level: 953,
        humidity: 90,
        temp_kf: 0,
      },
      weather: [
        {
          id: 800,
          main: "Clear",
          description: "clear sky",
          icon: "01n",
        },
      ],
      clouds: {
        all: 4,
      },
      wind: {
        speed: 0.94,
        deg: 94,
        gust: 0.96,
      },
      visibility: 10000,
      pop: 0,
      sys: {
        pod: "n",
      },
      dt_txt: "2021-11-25 00:00:00",
    },
  ],
  city: {
    id: 2867714,
    name: "Munich",
    coord: {
      lat: 48.1374,
      lon: 11.5755,
    },
    country: "DE",
    population: 1260391,
    timezone: 3600,
    sunrise: 1637389687,
    sunset: 1637422254,
  },
};
