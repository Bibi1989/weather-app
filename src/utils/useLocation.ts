import { useEffect, useState } from "react";

export const useLocation = () => {
  const [state, setState] = useState<number[]>([]);

  useEffect(() => {
    navigator.geolocation.getCurrentPosition(function (position) {
      setState([position.coords.longitude, position.coords.latitude]);
    });
  }, []);

  return state;
};
