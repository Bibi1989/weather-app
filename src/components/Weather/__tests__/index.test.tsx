import React from "react";
import { render } from "@testing-library/react";
import createMockStore from "redux-mock-store";

import WeatherComponent from "..";
import { FormatedWeatherInterface } from "typescript/weather.types";
import { Provider } from "react-redux";
import { formatWeatherReturn } from "utils/formatWeatherReturn";
import { mockWeatherData } from "utils/test-data/mockWeatherData";

const mockStore = createMockStore([]);

type Props = {
  weathers: FormatedWeatherInterface[];
  weatherLength: number;
  loading: boolean;
};

const renderWeatherComponent = (props: Partial<Props> = {}) => {
  const defaultProps: Props = {
    weathers: [],
    weatherLength: 5,
    loading: false,
  };

  const store = mockStore(props.weathers ?? []);

  return render(
    <Provider store={store}>
      <WeatherComponent {...defaultProps} {...props} />
    </Provider>
  );
};

describe("Weather component", () => {
  it("Weather component snapshot", () => {
    const { container } = renderWeatherComponent();

    expect(container).toMatchSnapshot();
  });

  it("should have the list weathers not greater than 1", async () => {
    let weatherLength = formatWeatherReturn(
      mockWeatherData.list,
      "metric"
    ).length;

    const { queryByTestId } = renderWeatherComponent({
      weatherLength,
      loading: false,
    });

    const weatherCard = queryByTestId("weatherCard");

    expect(weatherCard).not.toBeInTheDocument();
  });

  it("Weather card should be in the document", async () => {
    let weathers = formatWeatherReturn(mockWeatherData.list, "metric");
    let weatherLength = formatWeatherReturn(
      mockWeatherData.list,
      "metric"
    ).length;

    const { queryByTestId } = renderWeatherComponent({
      weathers,
      weatherLength,
      loading: false,
    });

    const weatherCard = queryByTestId("weatherCard");

    expect(weatherCard).toBeInTheDocument();
  });

  it("should render an empty component when weathers list is empty", async () => {
    let weathers: FormatedWeatherInterface[] = [];
    let weatherLength = weathers.length;

    const { getByTestId } = renderWeatherComponent({
      weathers,
      weatherLength,
      loading: false,
    });

    const emptyComponent = getByTestId("empty");

    expect(emptyComponent).toBeInTheDocument();
  });
});
