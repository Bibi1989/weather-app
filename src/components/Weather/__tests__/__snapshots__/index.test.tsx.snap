// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather component Weather component snapshot 1`] = `
<div>
  <div
    class="sc-iCfMLu hLoRyz"
  >
    <div
      class="sc-iqseJM GzWZU"
      data-testid="empty"
    >
      <p>
        Sorry there is/are no weathers forcast
      </p>
    </div>
  </div>
</div>
`;
